src/zenml/integrations/mlflow/model_registries/mlflow_model_registry.py

```diff
@@ -0,0 +1,796 @@
+        name: str,
+        description: Optional[str] = None,
+        tags: Optional[Dict[str, str]] = None,
+        model_source_uri: Optional[str] = None,
+        version: Optional[str] = None,
+        version_description: Optional[str] = None,
+        version_tags: Optional[Dict[str, str]] = None,
+        registry_metadata: Optional[Dict[str, str]] = None,
+        zenml_version: Optional[str] = None,
+        zenml_pipeline_run_id: Optional[str] = None,
+        zenml_pipeline_name: Optional[str] = None,
+        zenml_step_name: Optional[str] = None,
+        **kwargs: Any,
+    ) -> ModelVersion:
+        """Register a model version to the MLflow model registry.
+
+        Args:
+            name (str): The name of the model.
+            description (str, optional): The description of the model.
+            tags (Dict, optional): The tags of the model.
+            model_source_uri (str, optional): The source URI of the model.
+            version (str, optional): The version of the model.
+            version_description (str, optional): The description of the model
+                version.
+            version_tags (Dict, optional): The tags of the model version.
+            registry_metadata (Dict, optional): The registry metadata of the
+                model version.
+            zenml_version (str, optional): The ZenML version.
+            zenml_pipeline_run_id (str, optional): The ZenML pipeline run ID.
+            zenml_pipeline_name (str, optional): The ZenML pipeline name.
+            zenml_step_name (str, optional): The ZenML step name.
+            **kwargs: Additional keyword arguments.
+
+        Raises:
+            MlflowException: If the registered model does not exist.
+
+        Returns:
+            The registered model version.
+        """
+        # Check if the model exists, if not create it.
+        if not self.check_model_exists(name):
+            logger.warning(
+                f"No registered model with name {name} found. Creating a new"
+                "registered model."
+            )
+            self.register_model(
+                name=name,
+                description=description,
+                tags=tags,
+            )
+        try:
+            # Inform the user that the version is ignored.
+            if version:
+                logger.info(
+                    f"MLflow model registry does not take a version as an argument. "
+                    f"Registering a new version for the model `'{name}'` "
+                    f"a version will be assigned automatically."
+                )
+            # Set the tags.
+            if not version_tags:
+                version_tags = {}
+            version_tags["zenml_version"] = zenml_version or __version__
+            version_tags["zenml_pipeline_run_id"] = zenml_pipeline_run_id or ""
+            version_tags["zenml_pipeline_name"] = zenml_pipeline_name or ""
+            version_tags["zenml_step_name"] = zenml_step_name or ""
```

Comment:
Do we really need empty tag values for all of those? Seems like we can just leave them out if they don't get passed?